geth/api/connection.go

```diff
@@ -0,0 +1,65 @@
+package api
+
+import (
+	"fmt"
+)
+
+// ConnectionState represents device connection state and type,
+// as reported by mobile framework.
+//
+// Zero value represents default assumption about network
+// connection until first update — online, on cellular and not expensive.
+type ConnectionState struct {
+	Offline   bool           `json:"offline"`
+	Type      ConnectionType `json:"type"`
+	Expensive bool           `json:"expensive"`
+}
+
+// ConnectionType represents description of available
+// connection types as reported by React Native (see
+// https://facebook.github.io/react-native/docs/netinfo.html)
+// We're interested mainly in 'wifi' and 'cellular', but
+// other types are also may be used.
+type ConnectionType byte
+
+// NewConnectionType creates new ConnectionType from string.
+func NewConnectionType(s string) ConnectionType {
+	switch s {
+	case "cellular":
+		return ConnectionCellular
+	case "wifi":
+		return ConnectionWifi
+	}
+
+	return ConnectionUnknown
+}
+
+// ConnectionType constants
+const (
+	ConnectionCellular ConnectionType = iota // default value, LTE, 4G, 3G, EDGE, etc.
```

Comment:
I'd make this enum based on string instead
```go
type ConnectionType string

const (
       ConnectionCellular = ConnectionType("cellular")
       ConnectionOffline = ConnectionType("offline")
)
```

With this we can:
1) Avoid hardcoded strings in `switch`es and simplify `String()`
2) Make JSON PRC call a bit more readable w/o remembering which constant (0, 1, 2) means what.

Reply:
I implemented it with string type at the beginning, but the drawback here is that `ConnectionState` zero value is unusable. That's the main reason of switching to byte.

JSON-RPC is unaffected here, as it receives string anyway (and gets converted into `ConnectionType` in `NewConnectionType()` function).